src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java

```diff
@@ -368,77 +374,91 @@ public boolean logNewTimestamp() {
 
-      if (!isTimeSynchronized()) {
-        continue;
-      }
+  public static boolean info(String msg) {
+    return getInstance().logMessage(msg, DTLog.Level.INFO);
+  }
 
-      try {
-        INSTANCE = new DTLogWriter();
-        return;
-      } catch (IOException e) {
-        // ignore
-      }
+  public static boolean warn(String msg) {
+    return getInstance().logMessage(msg, DTLog.Level.WARN);
+  }
+
+  public static boolean error(String msg) {
+    return getInstance().logMessage(msg, DTLog.Level.ERROR);
+  }
+
+  private boolean logMessage(Supplier<String> msgSupplier, DTLog.Level logLevel) {
+    if (logLevel.ordinal() < level.ordinal()) {
+      return false;
     }
+
+    return logMessage(msgSupplier.get(), logLevel);
   }
 
-  private static boolean isTimeSynchronized() {
-    return DriverStation.isDSAttached() && LocalDate.now(Clock.systemUTC())
-                                                    .getYear() >= 2000;
+  public static boolean debug(Supplier<String> msgSupplier) {
+    return getInstance().logMessage(msgSupplier, DTLog.Level.DEBUG);
   }
 
-  public static DTLogWriter getInstance() {
-    return INSTANCE;
+  public static boolean info(Supplier<String> msgSupplier) {
+    return getInstance().logMessage(msgSupplier, DTLog.Level.INFO);
   }
 
-  private static int getTeamNumber() {
-    if (RobotBase.isSimulation()) {
-      return UNKNOWN_TEAM_NUMBER;
-    }
+  public static boolean warn(Supplier<String> msgSupplier) {
+    return getInstance().logMessage(msgSupplier, DTLog.Level.WARN);
+  }
 
-    Enumeration<NetworkInterface> interfaces;
-    try {
-      interfaces = NetworkInterface.getNetworkInterfaces();
-    } catch (SocketException e) {
-      return UNKNOWN_TEAM_NUMBER;
-    }
+  public static boolean error(Supplier<String> msgSupplier) {
+    return getInstance().logMessage(msgSupplier, DTLog.Level.ERROR);
+  }
 
-    // Use IP address 10.??.??.2 to obtain team number
-    while (interfaces.hasMoreElements()) {
-      NetworkInterface i = interfaces.nextElement();
-      Enumeration<InetAddress> addresses = i.getInetAddresses();
-      while (addresses.hasMoreElements()) {
-        String[] split = addresses.nextElement()
-                                  .getHostAddress()
-                                  .split("\\.");
-        if (!"10".equals(split[0]) || !"41".equals(split[3])) {
-          continue;
-        }
+  public static void logException(Throwable exception, DTLog.Level logLevel) {
+    getInstance().logMessage(exception.toString(), logLevel);
+    debug(() -> {
+      ByteArrayOutputStream debugOutput = new ByteArrayOutputStream();
+      PrintStream debugPrinter = new PrintStream(debugOutput);
+      exception.printStackTrace(debugPrinter);
+      String str = new String(debugOutput.toByteArray());
+      return str.substring(str.indexOf('\t'));
+    });
+  }
 
+  public static void init(DTLog.Level logLevel) {
+    while (true) {
+      if (!RobotController.isSystemTimeValid()) {
```

Comment:
I'm not clear on the logic here. It seems like you want to wait until the system time IS valid, then create and return the instance. It seems like the check may be inverted.

Reply:
Copy-paste error, will fix in a moment